### PR TITLE
UI polish for memory page. 

### PR DESCRIPTION
**Pull request by polina-c**
RELEASE_NOTE_EXCEPTION=[too small to mention]

Explain static and live references:
Screenshot 2023-02-21 at 2 12 54 PM

Add sign for delta values:
Screenshot 2023-02-21 at 1 58 52 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart [hunk 1]
```diff
@@ -117,6 +117,14 @@ class _InstanceColumn extends ColumnData<DiffClassStats> {
     }
   }
 
+  @override
+  String getDisplayValue(DiffClassStats classStats) {
+    // Add leading sign for delta values.
+    final value = getValue(classStats);
+    if (dataPart != _DataPart.delta || value <= 0) return value.toString();
+    return '+$value';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:125]: seems like we should also be adding the negative sign when the value is negative

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:125]: when value is negative, the sign is already here

File: packages/devtools_app/lib/src/shared/diagnostics/references.dart [hunk 2]
```diff
@@ -69,7 +69,7 @@ Future<void> addChildReferences(
 
       variable.addAllChildren([
         DartObjectNode.references(
-          'live',
+          'live (application in its current state)',
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/references.dart:72]: wording nit. maybe:

'live (objects currently alive in the application)"
"static (objects alive at the time of snapshot ${selection.heap.snapshotName})"

File: packages/devtools_app/test/test_infra/fixtures/memory_app/lib/main.dart [hunk 3]
```diff
@@ -35,11 +35,16 @@ class MyHomePage extends StatefulWidget {
...
   final _garbage = <_MyGarbage>[];
   int _counter = 0;
+  _MyGarbage _gcableItem = _MyGarbage(0);
 
   void _incrementCounter() {
     setState(() {
       _counter++;
       _garbage.add(_MyGarbage(0));
+      if (identityHashCode(_gcableItem) < 0) {
+        print('we need this line to use the field');
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/test_infra/fixtures/memory_app/lib/main.dart:45]: remove print

**Review by kenzieschmoll — APPROVED** [review]
could nits but lgtm

File: packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart [hunk 4]
```diff
@@ -117,6 +117,8 @@ class AdaptedHeapData {
 
   late DateTime created;
 
+  String snapshotName = '';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart:120]: should this be late final?

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart:120]: It is initialized after creation and i do not want anything to fail if it is accessed before initialization.

**Pull request merged**